dask/utils.py

```diff
@@ -2081,3 +2082,15 @@ def is_namedtuple_instance(obj: Any) -> bool:
         and hasattr(obj, "_fields")
         and hasattr(obj, "_field_defaults")
     )
+
+
+def get_default_shuffle_algorithm() -> str:
```

Comment:
It makes sense to replace the various `config.get("shuffle", None)` calls throughout the codebase with `get_default_shuffle_algorithm()`. However, we really do want to avoid using "disk" for the shuffle-based groupby code path. Otherwise we will definitely be introducing performance regressions for the threaded scheduler when `split_out>1`.

Perhaps the `get_default_shuffle_algorithm` can accept an optional argument to override the fallback shuffle-algorithm default? e.g. `get_default_shuffle_algorithm(fallback="tasks")`?

Reply:
I chose to keep these sections the way they are and left a comment. When switching to P2P eventually we'll revisit all of this again